src/librustc_attr/builtin.rs

```diff
@@ -908,11 +908,12 @@ pub fn find_repr_attrs(sess: &ParseSess, attr: &Attribute) -> Vec<ReprAttr> {
                     let parse_alignment = |node: &ast::LitKind| -> Result<u32, &'static str> {
                         if let ast::LitKind::Int(literal, ast::LitIntType::Unsuffixed) = node {
                             if literal.is_power_of_two() {
-                                // rustc::ty::layout::Align restricts align to <= 2^29
-                                if *literal <= 1 << 29 {
+                                // Many targets don't support global variables
+                                // with alignment greater than the page size.
```

Comment:
And also page size is always at last 4096, I take it? That is left implicit here, right now the comment doesn't entirely match what the code does.